_plugins/yaml_content_disposition.rb

```diff
@@ -0,0 +1,16 @@
+require 'webrick'
+
+module Jekyll
+  module Commands
+    class Serve
+      class Servlet < WEBrick::HTTPServlet::FileHandler
+        prepend(Module.new do
+          def do_GET(req, res)
+            res.header.merge!('Content-Disposition' => 'attachment') if req.path.end_with?('.yml')
```

Comment:
Jekyll [supports custom headers](https://jekyllrb.com/docs/configuration/webrick/), but not for specific file types. This plugin follows the pattern from [the code which introduced that option](https://github.com/jekyll/jekyll/pull/4224/files#diff-7a28653629164aecdb4936731e0b70992db66523b2f8849a6295715326689d7f) to target YAML files specifically.